keps/sig-api-machinery/5073-declarative-validation-with-validation-gen/README.md

```diff
@@ -948,6 +988,180 @@ The linter, as previously described, will enforce rules to address valid zero-va
 
 The linter will flag any violations of these rules, ensuring consistent zero-value handling and preventing related errors. This automated enforcement is crucial for catching issues early in the development process.
 
+### Referencing Fields in Validation-Gen For Cross-Field Validation Rules
+
+Cross-field validation refers to validation rules that depend on the values of multiple fields within a struct or across nested structs. Cross-field validations require a method for referencing additional fields from validation-gen IDL tags.
+
+#### Field Reference Strategies
+
+`validation-gen` supports two strategies for referencing fields in cross-field validations:
+
+1. **Field Paths** - Direct references using dot notation
+1. **Virtual Fields** - Identifier-based references for relationships
+
+##### Field Path Strategy
+
+Field paths use JSON field names with dot notation to reference other fields. 
+
+**When tags are placed on fields:**
+
+-  Field paths are relative to the parent typedef (allowing sibling access)
+-  No special `self` or `parent` keywords allowed
+-  Example: `siblingField` or `siblingStruct.childField`
+
+```go
+type Config struct {
+    MinValue int32 `json:"minValue"`
+    MaxValue int32 `json:"maxValue"`
+    
+    // +k8s:minimum(constraint: minValue)
+    // +k8s:maximum(constraint: maxValue)
+    Current int32 `json:"current"`
+}
+```
+
+**When tags are placed on the common ancestor typedef:**
+
+-  All references are to child fields
+
+```go
+// NOTE: this is illustrative - minimum/maximum tags do not support "target"
+// +k8s:minimum(target: current, constraint: minValue)
+// +k8s:maximum(target: current, constraint: maxValue)
+type Config struct {
+    MinValue int32 `json:"minValue"`
+    MaxValue int32 `json:"maxValue"`
+    Current int32 `json:"current"`
+}
+```
+
+##### Virtual Field Strategy
+
+Virtual fields provide identifier-based references for cross-field relationships. They are scoped to a GVK (GroupVersionKind) namespace.
+
+**Virtual Field Reference Tags:**
+
+-  `+k8s:reference(name: <name>)` - Creates a named reference to a field's value
+-  `+k8s:memberOf(group: <groupname>)` - Adds field to a group for union/mutual exclusion validation
+-  `+k8s:listMapItem(pairs: [[key,value],...])` - References specific items by key value in listType=map lists where the key(s) must include all of the list maps's keys
+
+```go
+type Config struct {
+    // +k8s:reference(name: minValue)
```

Comment:
Can we switch examples?

How about we use the union example for virtual fields, and we use the min/max example for field paths?

I think this better highlights our best practices.

If we ever do need min/max to support references we can add that, but it feels more natural to use the simple sibling reference approach for the common case.